plugin/restore/mysql/mysql.go

```diff
@@ -532,64 +542,109 @@ func (r *Restore) FetchArchivedBinlogFiles(ctx context.Context) error {
 	if err != nil {
 		return fmt.Errorf("failed to read local binlog files, error[%w]", err)
 	}
-	// build a local file size map from file name to size
-	localFileMap := make(map[string]int64)
-	for _, localFile := range binlogFilesLocalSorted {
-		localFileMap[localFile.Name] = localFile.Size
-	}
 
-	todo := make(map[string]bool)
-
-	for _, serverBinlog := range binlogFilesOnServerSorted {
-		// We don't download the latest binlog in SyncArchivedBinlogFiles()
-		if serverBinlog.Name == latestBinlogFileOnServer.Name {
-			continue
+	if len(binlogFilesLocalSorted) != 0 {
+		// Re-download inconsistent local binlog files.
+		// This is a best-effort task, because the corresponding binlog files may not exist on server.
+		log.Debug("Re-downloading inconsistent local binlog files")
```

Comment:
We only re-download the binlog files whose size does not match. I think we only display this log when we really need to re-download some binlog files.

Reply:
It's for debugging. The purpose is to say that we are at this stage and only log once. IMO it's OK.